test/test_client_async.py

```diff
@@ -393,13 +393,34 @@ def test_maybe_refresh_metadata_cant_send(mocker, client):
     assert not client._metadata_refresh_in_progress
 
 
-def test_schedule():
-    pass
-
-
-def test_unschedule():
-    pass
-
+def test_schedule(mocker, client):
+    task_delay_secs = 2
+    tolerance_secs = 4
+    sleep_secs = 0.5
+    task_result = 'task_succeeded'
+    def task():
+        return task_result
+    start_time = time.time()
+    future = client.schedule(task, start_time + task_delay_secs)
+    while not future.is_done:
+        time.sleep(sleep_secs)
+        client.poll(timeout_ms=100, delayed_tasks=True)
+        assert time.time() < start_time + tolerance_secs
```

Comment:
Can we add parens to make the order of operations more obvious?

    time.time() < (start_time + tolerance_secs)

Reply:
Sure